repos.d/ros.yaml

```diff
@@ -0,0 +1,56 @@
+###########################################################################
+# ROS
+###########################################################################
+{% macro ros(version, name, minpackages, valid_till) %}
+- name: ros_{{version}}_{{name}}
+  type: repository
+  desc: ROS {{name}} {% if version > 1 %}(ROS {{version}}){% endif %}
+  statsgroup: ros
+  family: ros
+  ruleset: ros
+  color: '22314E'
+  minpackages: {{minpackages}}
+  {% if valid_till %}
+  valid_till: {{valid_till}}
+  {% endif %}
+  default_maintainer: fallback-mnt-ros@repology
+  sources:
+    - name: distribution.yaml
+      fetcher:
+        class: FileFetcher
+        url: https://raw.githubusercontent.com/ros/rosdistro/refs/heads/master/{{name}}/distribution.yaml
+      parser:
+        class: RosYamlParser
+  repolinks:
+    - desc: ROS home
+      url: https://ros.org/
+    - desc: ROS packages
+      url: https://index.ros.org/
+    - desc: {{name}} GitHub repository
+      url: https://github.com/ros/rosdistro/tree/master/{{name}}
+    - desc: {{name}} home
+      url: https://{% if version > 1 %}docs.ros.org/en{% else %}wiki.ros.org{% endif %}/{{name}}
+  groups: [ all, production, ros, ros{{version}} ]
+{% endmacro %}
+
+{{ ros(1, 'groovy', minpackages=200, valid_till='2014-07-31') }}
```

Comment:
For the note, it may make sense not to add EoLed repos of that would simplify code (e.g. by dropping legacy conditions for them).

Reply:
I did not have to write anything specific for these old things, and it seem to work so I included them. But I don't really need those.